http/http/src/main/java/io/helidon/http/HeaderValueIterable.java

```diff
@@ -55,18 +45,14 @@ private int getIterableSize() {
 
     @Override
     public List<String> allValues() {
-        if (values == null) {
-            values = new ArrayList<>(getIterableSize());
-            originalValues.forEach(values::add);
-        }
-        return values;
+        return StreamSupport.stream(originalValues.spliterator(), false).toList();
     }
 
     @Override
     public int valueCount() {
-        if (values == null) {
-            return getIterableSize();
+        if (valueCount == 0) {
```

Comment:
Although this new incarnation of `HeaderValueIterable` does not _itself_ permit values to be added to the header, it's possible that the underlying `Iterable` that we are wrapping _might_ change in size or contents over time. It's probably very unlikely, but it's possible.

With that in mind, `valueCount()` might need to iterate over the iterable each time it is invoked.

The potential changeability over time of the underlying `Iterable` is another good reason to _not_ maintain a copy of the values inside the class. I know the implementation no longer does that but I'm just reinforcing the idea.

Reply:
Updated valueCount() to perform the count on every invocation to take into account underlying Iterables that has possibility to mutate. Also added a test to cover this scenario, i.e. to check that valueCount() and allValues() still work as expected even if the underlying Iterable has changed.

Reply:
We should never wrap a mutable object. If the object is possibly mutable, we should create an immutable copy